src/main/java/com/google/cloud/tools/appengine/experimental/AppEngineRequest.java

```diff
@@ -16,15 +16,14 @@
 
 package com.google.cloud.tools.appengine.experimental;
 
-import java.util.concurrent.Future;
+import com.google.cloud.tools.appengine.experimental.internal.cloudsdk.CloudSdkRequestFuture;
+
+import java.io.InputStream;
 
 /**
  * Interface for remote App Engine Requests.
  * @param <T> return request result type
  */
 public interface AppEngineRequest<T> {
```

Comment:
Question: does a client need parameterization on this class? Looking at the example code below, seems like clients will use `AppEngineRequestFactory`, which returns `AppEngineRequest<DeployResult>`. Are there uses cases where clients want somethings else than `DeployResult`? If so, will client by-pass the factory?